src/bellpepper/r1cs.rs

```diff
@@ -85,29 +86,66 @@ macro_rules! impl_nova_shape {
             &constraint.2,
           );
         }
-
         assert_eq!(num_cons_added, num_constraints);
 
+        A.cols = num_vars + num_inputs;
+        B.cols = num_vars + num_inputs;
+        C.cols = num_vars + num_inputs;
+
         let S: R1CSShape<G> = {
-          // Don't count One as an input for shape's purposes.
-          let res = R1CSShape::new(num_constraints, num_vars, num_inputs - 1, &A, &B, &C);
+          let res = R1CSShape::new(num_constraints, num_vars, num_inputs - 1, A, B, C);
           res.unwrap()
         };
-
         S
       }
     }
   };
 }
 
-impl_nova_shape!(ShapeCS);
+impl<G: Group> NovaShape<G> for ShapeCS<G>
```

Comment:
I may be missing something: could we fold this impl in a call of the `impl_nova_shape` macro?